src/main/java/at/hannibal2/skyhanni/features/mining/FlowstateHelper.kt

```diff
@@ -0,0 +1,173 @@
+        if (hasFlowstate() == null) return
+
+        pickobulusPattern.matchMatcher(event.message) {
+            streakEndTimer = SimpleTimeMark.now().plus(10.seconds)
+            blockBreakStreak += group("amount").toInt()
+            createDisplay()
+            attemptClearDisplay()
+            displayDirty = true
+        }
+    }
+
+    @SubscribeEvent
+    fun onTick(event: LorenzTickEvent) {
+        if (!MiningAPI.inCustomMiningIsland()) return
+
+        attemptClearDisplay()
+    }
+
+    private fun attemptClearDisplay() {
+        if (streakEndTimer.isInFuture()) return
+        blockBreakStreak = 0
+        displayDirty = true
+        createDisplay()
+    }
+
+    @SubscribeEvent
+    fun onRenderOverlay(event: GuiRenderEvent.GuiOverlayRenderEvent) {
+        if (!MiningAPI.inCustomMiningIsland()) return
+        if (!config.flowstateHelper) return
+        if (hasFlowstate() == null) return
+
+        if (display == null || streakEndTimer.isInFuture()) createDisplay()
+        display?.let {
+            config.flowstateHelperPosition.renderRenderables(
+                it,
+                1,
+                "Flowstate Helper",
+                true
+            )
+        }
+    }
+
+    private fun createDisplay() {
+        if (displayDirty) {
+            displayDirty = false
+            createDisplayTimer()
+            createDisplayBlock()
+            createDisplaySpeed()
+        } else {
+            createDisplayTimer()
+        }
+        display = config.flowstateHelperAppearance.map { it.renderable }
+    }
+
+    private fun createDisplayTimer(): Renderable {
+        var timeRemaining = streakEndTimer.minus(SimpleTimeMark.now()).inPartialSeconds
+        if (timeRemaining < 0.0) timeRemaining = 0.0
+
+        FlowstateElements.TIMER.renderable = Renderable.string("Time Remaining: §b$timeRemaining")
+        return FlowstateElements.TIMER.renderable
+    }
+
+    private fun createDisplayBlock(): Renderable {
+        val textColor = if (blockBreakStreak < 200) "§7" else "§f"
+        FlowstateElements.STREAK.renderable = Renderable.string("Streak: $textColor$blockBreakStreak")
+        return FlowstateElements.STREAK.renderable
+    }
+
+    private fun createDisplaySpeed(): Renderable {
+        FlowstateElements.SPEED.renderable = Renderable.string("§6+${getSpeedBonus()}⸕")
+        return FlowstateElements.SPEED.renderable
+    }
+
+    private fun getSpeedBonus(): Int {
+        val flowstateLevel = hasFlowstate() ?: 0
+        if (blockBreakStreak >= 200) return 200 * flowstateLevel
+        return blockBreakStreak * flowstateLevel
+    }
+
+    @SubscribeEvent
+    fun onChangeItem(event: ItemInHandChangeEvent) {
+        hasChangedItem = true
+    }
+
+    @SubscribeEvent
+    fun onIslandChange(event: IslandChangeEvent) {
+        blockBreakStreak = 0
+        displayDirty = true
+        streakEndTimer = SimpleTimeMark.now()
+        createDisplay()
```

Comment:
could you just set the streak end timer to now (or probably better farPast) then call attemptClearDisplay()